Fix : Remove all the warning to compile the code without any.

sagar-barapatre opened the pull request:
### Description

There were a total of 16 warnings when the code was compiled locally. The main purpose of this PR was to remove all the warnings before hand so that the code compiles without any such warnings.


Fixes #67 

### Type of Change: bug


- Code
- Quality Assurance
- Outreach

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)



### How Has This Been Tested?
![Screenshot 2021-01-28 at 9 21 43 AM](https://user-images.githubusercontent.com/64516304/106087566-46e7c280-614a-11eb-923e-d3ccdcb48f29.png)



### Checklist:


- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] I have written Kotlin Docs whenever is applicable


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

sagar-barapatre committed:
Updated store.js

Removed all the warnings while compiling this file

sagar-barapatre committed:
Updated Private Routes.js file

Removed all the warnings occuring while compiling this file

sagar-barapatre committed:
Updated AuthRoute.js

Removed the warnings to compile this file warning less.

sagar-barapatre committed:
Update form.js

Removed the warnings in this file to run the file without any warnings

sagar-barapatre committed:
Update login.js

Removed the warnings on this page too.

sagar-barapatre committed:
Update login.js

sagar-barapatre committed:
Update Menubar.js

sagar-barapatre committed:
Update Navbar.js

sagar-barapatre committed:
Update Profile.js

sagar-barapatre committed:
Update PublishedForm.js

sagar-barapatre committed:
Update Register.js

sagar-barapatre committed:
Update UnpublishedForm.js

sagar-barapatre committed:
Update Navbar.js

isabelcosta commented:
@sagar-barapatre can you please fix the merge conflicts here?

sagar-barapatre committed:
Merge branch 'develop' into develop

sagar-barapatre commented:
@isabelcosta I have fixed the conflicts. Can you please take a look.

Amulya-coder commented:
@sagar-barapatre please squash your commits into one. It's always better to have one commit per change to keep the history clean.

khushishikhu commented:
@sagar-barapatre  You should change the PR title according to AnitaB.org's guidelines

khushishikhu reviewed (commented):
@sagar-barapatre You should change the PR title according to the AnitaB.org's guidelines and also the commit message style

sagar-barapatre committed:
fix: a bug fix

Remove the unnecessary variables used in the code to remove the warnings

sagar-barapatre committed:
fix : remove the unnecessary variables used

sagar-barapatre committed:
fix : remove unnecessary variables used

sagar-barapatre committed:
fix : remove extra variable used

sagar-barapatre committed:
fix : use setState() to mutate the props

sagar-barapatre committed:
fix : use setState() to mutate the props

sagar-barapatre committed:
fix : remove the unnecessary import in header

sagar-barapatre committed:
fix : remove the imports that are not required

sagar-barapatre committed:
fix : change payload value

Payload value should be assigned with "res.data" instead of "res.response.data"

sagar-barapatre committed:
fix : change payload value to make it accessible 

Payload value is correctly accessed by changing the payload value from "id" to "res.id"

sagar-barapatre committed:
fix : add a comment to remove the required warning

sagar-barapatre committed:
fix : add a comment in top to remove warning

sagar-barapatre commented:
@khushishikhu I have changed the commit message style but I am unable to figure out what should be the proper PR title

jalajcodes commented:
> @khushishikhu I have changed the commit message style but I am unable to figure out what should be the proper PR title. I didn't found it in the [Contributing Guidelines](https://github.com/anitab-org/open-source-programs-web/blob/develop/.github/PULL_REQUEST_TEMPLATE.md)

Adding a prefix just like the commit messages should do it.

sagar-barapatre commented:
@isabelcosta Please have a look now !

isabelcosta commented on src/components/Login.js:
```diff
@@ -1,12 +1,12 @@
+import { postLogin } from '../actions/login';
+import { Form, Grid, Image, Divider, Icon, Message, Button } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+// import login from './../styles/Login.css'
```
is this not needed?

sagar-barapatre replied:
The import login wasn't used anywhere in Login.js file which resulted in the warning "Used of undeclared variable login".
That's why we need to comment out that.

jalajcodes replied:
Have you tried rewriting it like this `import './../styles/Login.css'` ?

sagar-barapatre replied:
No

isabelcosta commented on src/components/Profile.js:
```diff
@@ -59,9 +59,18 @@ class Profile extends Component {
+                update: false
+            })
         }
+            // userinfo && userinfo.length === 1 ? this.state.update = true : this.state.update = false
```
what is this lane about 🤔

sagar-barapatre replied:
This warning was like we have to use setState() to mutate the props so I commented out this code and use the setState() to remove the warning

jalajcodes replied:
@sagar-barapatre I don't think its a good idea to use `setState` inside `render`. It [not advisable](https://stackoverflow.com/a/48226704/5182477)

This code should be modified in a way so that we don't have to mutate state inside render. For now I think you should revert the changes and disable eslint to supress the warnings. We can create another issue to fix it.

sagar-barapatre replied:
In warning it was specifically mentioned to use setState() to remove that

isabelcosta commented on src/components/Navbar.js:
```diff
@@ -42,6 +43,11 @@ class Navbar extends Component {
         const { isLoggedIn } = this.state
+       
+
+//    let loginButton = localStorage.getItem('token') !== null? this.state.isLoggedIn = true: this.state.isLoggedIn = false;
+//    this.setState({loginButton: loginButton});
```
should this be here?

sagar-barapatre replied:
This is also the same, warning was to use setState() instead of props to mutate it. So commenting that piece of code and writing the new one did workout for me.

isabelcosta replied:
Ok so you re-implemented the commented code? if so, then you can remove the commented code @sagar-barapatre

sagar-barapatre replied:
@isabelcosta Yeah done !!

sagar-barapatre committed:
fix : removed the commented out code.

jalajcodes commented on src/components/Menubar.js:
```diff
@@ -1,11 +1,11 @@
+import PropTypes from 'prop-types';
+import { Menu } from 'semantic-ui-react';
+import { dashboard, forms, submission } from '../urls';
+// import menubar from './../styles/Menubar.css'
```
same here

sagar-barapatre replied:
No, I just commented out that code because it was never used in that file.

jalajcodes commented on src/store.js:
```diff
@@ -1,14 +1,11 @@
 
 export const history = createBrowserHistory();
 
-const createStoreWithMiddleware = applyMiddleware(apiMiddleware)(createStore);
```
why did you remove that line?

sagar-barapatre replied:
because "createStoreWithMiddleware" was declared here and this variable was also never used in that file. So unnecessary declaration of variable was causing error.

jalajcodes reviewed (commented):
Please checkout my comments below

Rahulm2310 commented:
@sagar-barapatre Please don't use `develop` branch for submitting a pull request. Always use a seperate/new branch. It would be great if you could close this PR and open a new PR from a seperate branch. Thanks 🙂   

Cc: @isabelcosta

codesankalp commented:
Also, I think all the warnings have been resolved in this PR https://github.com/anitab-org/open-source-programs-web/pull/94 @isabelcosta. So should I close this?

codesankalp commented:
Closing this PR as it is fixed after adding QA-Checks and linting.
